cmd/ftl/cmd_init.go

```diff
@@ -38,6 +38,9 @@ func (i initGoCmd) Run(ctx context.Context, parent *initCmd) error {
 	if i.Name == "" {
 		i.Name = filepath.Base(i.Dir)
 	}
+	if strings.Contains(i.Name, "-") {
```

Comment:
This would still allow other invalid characters, so let's add a `backend/schema.ValidateIdentifier(n)` which uses a regex to validate that it's a valid identifier.

Reply:
I realise I should have been more specific in my comment on that ticket :) - what I should have said is that "modules must be valid identifiers"

Reply:
updated!